modules/compute/gke-node-pool/main.tf

```diff
@@ -367,6 +367,22 @@ resource "google_container_node_pool" "node_pool" {
       condition     = !(var.num_node_pools < 0 || var.num_slices < 0)
       error_message = "Negative integer value of num_node_pools or num_slices is not valid. Please use a positive integer value to set num_node_pools for CPUs and GPUS, and num_slices for TPUs."
     }
+    precondition {
+      condition     = var.enable_flex_start == true ? (var.auto_repair == false) : true
+      error_message = "enable_flex_start needs node auto_repair set to false."
+    }
+    precondition {
+      condition     = var.enable_flex_start == true ? (var.static_node_count == null) : true
```

Comment:
Noted that this is be true for now, but this won't always be a restriction. 

Discuss with the team to decide whether to add this precondition.